### PR TITLE
Fscloud

### PR DESCRIPTION
**Pull request by rajatagarwal-ibm**
### Description

* FSCloud profile implementation for Cloudant.
* Bug fix in cloudant module to provide ipv4 addresses to the subnet.
* Removed Dedicated example

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Review by ocofaigh — CHANGES_REQUESTED** [review]
You should use `modules/fscloud` instead of a `profiles` folder

File: examples/fscloud/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Financial Services Cloud profile example
+
+An end-to-end example that uses the [Profile for IBM Cloud Framework for Financial Services](../../modules/fscloud/) to deploy a Cloudant instance.
+
+The example uses the IBM Cloud Terraform provider to create the following infrastructure:
+
+- A resource group, if one is not passed in.
+- A cloudant instance
```
Comment (ocofaigh) [inline examples/fscloud/README.md:8]: A dedicated cloudant instance

File: modules/fscloud/variables.tf [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+variable "instance_name" {
+  description = "The name of the Cloudant instance"
+  type        = string
+}
+
+variable "legacy_credentials" {
+  type        = bool
+  description = "Use both legacy credentials, in addition to IAM credentials for authentication. If set to false, use use only IAM credentials."
+  default     = false # uses only IAM credentials
```
Comment (ocofaigh) [inline modules/fscloud/variables.tf:31]: hmm, we may need to enforce IAM credentials for fscloud - we should find out

File: modules/fscloud/variables.tf [hunk 3]
```diff
@@ -0,0 +1,84 @@
+variable "region" {
+  type        = string
+  description = "The IBM Cloud region where the Cloudant instance will be provisioned."
+  default     = "us-south"
+}
+
+variable "plan" {
```
Comment (ocofaigh) [inline modules/fscloud/variables.tf:7]: remove plan from here - it has to be standard for dedicated. hard code it in the main.tf of fscloud profile

File: modules/fscloud/outputs.tf [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  description = "A number of blocks of throughput units"
+  value       = module.cloudant.capacity
+}
+
+output "resource_keys_url" {
+  description = "The relative path to the resource keys for the instance"
+  value       = module.cloudant.resource_keys_url
+}
+
+output "public_endpoint" {
```
Comment (ocofaigh) [inline modules/fscloud/outputs.tf:55]: there wont be a public endpoint when we hard code the instance to private

File: examples/fscloud/version.tf [hunk 5]
```diff
@@ -2,11 +2,11 @@ terraform {
   required_version = ">= 1.0.0, <1.6.0"
 
   # Ensure that there is always 1 example locked into the lowest provider version of the range defined in the main
-  # module's version.tf (basic example), and 1 example that will always use the latest provider version (this example).
+  # module's version.tf (this example), and 1 example that will always use the latest provider version (complete + dedicated examples).
```
Comment (ocofaigh) [inline examples/fscloud/version.tf:5]: dedicated example is removed, so update to fscloud example

File: examples/fscloud/outputs.tf [hunk 6]
```diff
@@ -47,6 +52,16 @@ output "resource_keys_url" {
   value       = module.create_cloudant.resource_keys_url
 }
 
+output "public_endpoint" {
```
Comment (ocofaigh) [inline examples/fscloud/outputs.tf:55]: fscloud wont have public endpoint

**Review by ocofaigh — CHANGES_REQUESTED** [review]
See comments

File: tests/pr_test.go [hunk 7]
```diff
@@ -60,3 +60,29 @@ func TestRunUpgradeExample(t *testing.T) {
...
 }
+
+func TestRunFSCloudExample(t *testing.T) {
+	t.Parallel()
+
+	options := testhelper.TestOptionsDefaultWithVars(&testhelper.TestOptions{
+		Testing:      t,
+		TerraformDir: "examples/fscloud",
+		Prefix:       "fscloud",
+		Region:       "us-south", // For FSCloud locking into us-south since that is where the HPCS permanent instance is
```
Comment (ocofaigh) [inline tests/pr_test.go:71]: `the HPCS permanent instance is` -> `the dedicated host is provisioned`

File: tests/pr_test.go [hunk 8]
```diff
@@ -60,3 +60,29 @@ func TestRunUpgradeExample(t *testing.T) {
...
+		/*
+		 Comment out the 'ResourceGroup' input to force this tests to create a unique resource group to ensure tests do
+		 not clash. This is due to the fact that an auth policy may already exist in this resource group since we are
+		 re-using a permanent HPCS instance. By using a new resource group, the auth policy will not already exist
+		 since this module scopes auth policies by resource group.
+		*/
+		//ResourceGroup: resourceGroup,
```
Comment (ocofaigh) [inline tests/pr_test.go:78]: There is no HPCS used here, so you should be able to uncomment this

File: modules/fscloud/variables.tf (hunk 2 above)
Comment (ocofaigh) [reply modules/fscloud/variables.tf:20]: We should find this out before we create the DA. Can you ask in our channel perhaps?

**Review by ocofaigh — CHANGES_REQUESTED** [review]
As far as I remember, one of the reasons we only ran the dedicated environment test in other_test.go was because of the cost. Can you confirm what the cost is here before we start running it as part of every PR?
Also left some final minor comments

**Comment by rajatagarwal-ibm** [comment]
For the minimum configuration for the dedicated cloudant instance, the cost is about $86 per month. 

image

File: modules/fscloud/variables.tf (hunk 2 above)
Comment (rajatagarwal-ibm) [reply modules/fscloud/variables.tf:20]: Checked in the channel. We can go ahead just with IAM for FSCloud module.

**Pull request merged**